docs/tasks/manage-gpus/scheduling-gpus.md

```diff
@@ -4,154 +4,203 @@ approvers:
-  * You can specify GPU `limits` without specifying `requests` because Kubernetes
-    will use the limit as the request value by default.
-  * You can specify GPU in both `limits` and `requests` but these two values must equal.
+  * You can specify GPU `limits` without specifying `requests` because
+    Kubernetes will use the limit as the request value by default.
+  * You can specify GPU in both `limits` and `requests` but these two values
+    must equal.
   * You cannot specify GPU `requests` without specifying `limits`.
-- Containers (and pods) do not share GPUs.
-- Each container can request one or more GPUs.
-- It is not possible to request a portion of a GPU.
-- Nodes are expected to be homogenous, i.e. run the same GPU hardware.
-
-If your nodes are running different versions of GPUs, then use Node Labels and Node Selectors to schedule pods to appropriate GPUs.
-Following is an illustration of this workflow:
+- Containers (and pods) do not share GPUs. There's no overcommitting of GPUs.
+- Each container can request one or more GPUs. It is not possible to request a
+  fraction of a GPU.
 
-As part of your Node bootstrapping, identify the GPU hardware type on your nodes and expose it as a node label.
+Unlike with `alpha.kubernetes.io/nvidia-gpu`, when using `nvidia.com/gpu` as
+the resource, you don't have to mount any special directories in your pod
+specs. The device plugin is expected to inject them automatically in the
+container.
 
-```shell
-NVIDIA_GPU_NAME=$(nvidia-smi --query-gpu=gpu_name --format=csv,noheader --id=0 | sed -e 's/ /-/g')
-source /etc/default/kubelet
-KUBELET_OPTS="$KUBELET_OPTS --node-labels='alpha.kubernetes.io/nvidia-gpu-name=$NVIDIA_GPU_NAME'"
-echo "KUBELET_OPTS=$KUBELET_OPTS" > /etc/default/kubelet
-```
-
-Specify the GPU types a pod can use via [Node Affinity](/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity) rules.
+Here's an example:
 
 ```yaml
-kind: pod
 apiVersion: v1
+kind: Pod
 metadata:
-  annotations:
-    scheduler.alpha.kubernetes.io/affinity: >
-      {
-        "nodeAffinity": {
-          "requiredDuringSchedulingIgnoredDuringExecution": {
-            "nodeSelectorTerms": [
-              {
-                "matchExpressions": [
-                  {
-                    "key": "alpha.kubernetes.io/nvidia-gpu-name",
-                    "operator": "In",
-                    "values": ["Tesla K80", "Tesla P100"]
-                  }
-                ]
-              }
-            ]
-          }
-        }
-      }
+  name: cuda-vector-add
 spec:
+  restartPolicy: OnFailure
   containers:
-    -
-      name: gpu-container-1
+    - name: cuda-vector-add
+      image: "gcr.io/google_containers/cuda-vector-add:v0.1"
       resources:
         limits:
-          alpha.kubernetes.io/nvidia-gpu: 2
+          nvidia.com/gpu: 1 # requesting 1 GPU
+```
+
+### Deploying NVIDIA GPU device plugin
+
+There are currently two device plugin implementations for NVIDIA GPUs:
```

Comment:
>two device plugin implementations

two sample device plugin implementations ?

Reply:
To me, "sample" implies that they are not ready to be used as is and require some modification. However, both these implementations are ready to be used as-is.